packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -486,94 +509,99 @@ export class RunningSummarizer implements IDisposable {
      * Generates summary and listens for broadcast and ack/nack.
      * Returns true for ack, false for nack, and undefined for failure or timeout.
      * @param reason - reason for summarizing
-     * @param safe - true to generate summary in safe mode
+     * @param options - refreshLatestAck to fetch summary ack info from server,
+     * fullTree to generate tree without any summary handles even if unchanged
      */
-    private async summarize(reason: string, safe: boolean): Promise<boolean | undefined> {
+    private async summarize(
+        reason: string,
+        options: Omit<IGenerateSummaryOptions, "summaryLogger">,
+    ): Promise<boolean | undefined> {
         this.summarizeTimer.start();
 
         try {
-            return await this.summarizeCore(reason, safe);
-        } finally {
-            this.summarizeTimer.clear();
-            this.pendingAckTimer.clear();
-        }
-    }
-
-    private async summarizeCore(reason: string, safe: boolean): Promise<boolean | undefined> {
-        ++this.summarizeCount;
-
-        // Wait to generate and send summary
-        const summaryData = await this.generateSummaryWithLogging(reason, safe);
-        this.heuristics.recordAttempt(summaryData?.referenceSequenceNumber);
-        if (!summaryData || !summaryData.submitted) {
-            // Did not send the summary op
-            this.raiseSummarizingError("Error while generating or submitting summary");
-            return undefined;
-        }
+            ++this.summarizeCount;
+
+            // Wait to generate and send summary
+            const summaryData = await this.generateSummaryWithLogging(reason, options);
+            this.heuristics.recordAttempt(summaryData?.referenceSequenceNumber);
+            if (!summaryData || !summaryData.submitted) {
+                // Did not send the summary op
+                this.raiseSummarizingError("Error while generating or submitting summary");
```

Comment:
Since you are poking at all this code - any chance we can replace summaryData.submitted to summaryData.error (negative) and force layers provide actual error object that goes into telemetry and events?

Reply:
I have changed this in the most recent commit.